### ISSUE TITLE
Consul monitor client freezes terminal when no messages are pending

### ISSUE THREAD

**Issue by MichaelSteffens** [hashicorp/consul#3317]
When running "consul monitor" while there are no messages due for delivery, the client terminal freezes. No signals are accepted. The consul executable can neither be terminated (CTRL-C) nor stopped (CTRL-Z).

The freeze persists until at least one message is printed. You can trigger this for example by restarting another cluster member. From there on "consul monitor" client reacts on signals as expected.

I'm actually observing this behavior since 0.8.0. Until 0.7.5 it might also have been present, but didn't hurt, because these always include the client RPC connection in the log:
```
[INFO] agent.rpc: Accepted client: 127.0.0.1:38092
```
thereby unlocking the client. This is no longer done since 0.8.0.

### `consul version` for both Client and Server
Client: `0.9.0`
Server: `0.9.0`

### `consul info` for both Client and Server
Client:
```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease = 
        revision = b79d951
        version = 0.9.0
consul:
        bootstrap = false
        known_datacenters = 2
        leader = false
        leader_addr = 192.168.4.33:8300
        server = true
raft:
        applied_index = 68385
        commit_index = 68385
        fsm_pending = 0
        last_contact = 72.966225ms
        last_log_index = 68385
        last_log_term = 11
        last_snapshot_index = 65552
        last_snapshot_term = 9
        latest_configuration = [{Suffrage:Voter ID:192.168.4.31:8300 Address:192.168.4.31:8300} {Suffrage:Voter ID:192.168.4.32:8300 Address:192.168.4.32:8300} {Suffrage:Voter ID:192.168.4.33:8300 Address:192.168.4.33:8300}]
        latest_configuration_index = 8
        num_peers = 2
        protocol_version = 2
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Follower
        term = 11
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 87
        max_procs = 2
        os = linux
        version = go1.8.3
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 8
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 2
        members = 3
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 249
        members = 4
        query_queue = 0
        query_time = 1
```

Server: see client

### Operating system and Environment details

Ubuntu Server 16.04, amd64

**Comment by C0reFast** [comment]
```
curl -v "127.0.0.1:8500/v1/agent/monitor?loglevel=err"                                                                                           
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8500 (#0)
> GET /v1/agent/monitor?loglevel=err HTTP/1.1
> Host: 127.0.0.1:8500
> User-Agent: curl/7.56.1
> Accept: */*
> 
^C
```

The agent didn't response a http header, so client is freezed to reading header.
I created a pull request https://github.com/hashicorp/consul/pull/3695 to solve this problem

**Comment by banks** [comment]
Fixed in #3891, released in 1.0.7. Thanks for the report and PR!
